elyra/pipeline/validation.py

```diff
@@ -422,23 +419,38 @@ async def _validate_custom_component_node_properties(self,
                     # Any component property with type `InputPath` will be a dictionary of two keys
                     # "value": the node ID of the parent node containing the output
                     # "option": the name of the key (which is an output) of the above referenced node
-                    if not isinstance(node_param_value, dict) or \
-                            len(node_param_value) != 2 or \
-                            set(node_param_value.keys()) != {'value', 'option'}:
+                    if not isinstance(node_param, dict) or len(node_param) != 2 or \
+                            set(node_param.keys()) != {'value', 'option'}:
                         response.add_message(severity=ValidationSeverity.Error,
                                              message_type="invalidNodeProperty",
                                              message="Node has malformed `InputPath` parameter structure",
                                              data={"nodeID": node.id,
                                                    "nodeName": node.label})
                     node_ids = list(x.get('node_id_ref', None) for x in node.component_links)
                     parent_list = self._get_parent_id_list(pipeline_definition, node_ids, [])
-                    if node_param_value.get('value') not in parent_list:
+                    node_param_value = node_param.get('value')
+                    if node_param_value not in parent_list:
                         response.add_message(severity=ValidationSeverity.Error,
                                              message_type="invalidNodeProperty",
                                              message="Node contains an invalid inputpath reference. Please "
                                                      "check your node-to-node connections",
                                              data={"nodeID": node.id,
                                                    "nodeName": node.label})
+                elif isinstance(node_param, dict) and node_param.get('activeControl') == 'NestedEnumControl':
+                    # TODO: Update this hardcoded check for xcom_push. This parameter is specific to a runtime
+                    # (Airflow). i.e. abstraction for byo validation?
+                    node_param_value = node_param['NestedEnumControl'].get('value')
+                    xcom_node = pipeline_definition.get_node(node_param_value)
+                    xcom_param = xcom_node.get_component_parameter('xcom_push')
+                    xcom_value = xcom_param.get('BooleanControl')
+                    if not xcom_value:
+                        response.add_message(severity=ValidationSeverity.Error,
+                                             message_type="invalidNodeProperty",
+                                             message="Node contains an invalid inputpath reference. The parent "
```

Comment:
NIT: I think referencing `inputpath` might be confusing to people that only use Airflow. May be worth changing it from `inputpath` to `input`?

I'm happy to make the change if we intend to try to get this merged today and others agree

Reply:
How about `input source reference`?